nornir/plugins/inventory/netbox.py

```diff
@@ -36,17 +36,41 @@ def __init__(
         )
         headers = {"Authorization": "Token {}".format(nb_token)}
 
+        def make_request(url: str) -> Any:
+            req = requests.get(url=url, headers=headers, params=filter_parameters)
+            req.raise_for_status()
+
+            if req.ok:
+                return req.json()
+
+        def req_all(url: str) -> Any:
+            req = make_request(url)
+            if isinstance(req, dict) and req.get("results") is not None:
```

Comment:
Fixing the `make_request` function will save you this check

Reply:
Not sure about that as this also checks for invalid `token` as well.

Reply:
an error shouldn't be signalled by returning `None` so `make_requests` should either succeed or return an exception. Also, this function doesn't seem to propagate the "error" if `req.get("results")` is `None` so at this point the issue seems to fail silently. If that's the desired behavior it should be correctly modeled by catching the expected error.